src/main/scala/io/circe/yaml/parser/Parser.scala

```diff
@@ -1,84 +0,0 @@
-import org.yaml.snakeyaml.Yaml
-import org.yaml.snakeyaml.nodes._
-
-object Parser {
-
-  object ScalarTag {
-    def unapply(tag: Tag) = if(tag.startsWith(Tag.PREFIX))
-      Some(tag.getClassName)
-    else
-      None
-  }
-
-  object CustomTag {
-    def unapply(tag: Tag) = if(!tag.startsWith(Tag.PREFIX))
-      Some(tag.getValue)
-    else
-      None
-  }
-
-  // Typecasing.  Because Java :(
-  private def convertYamlNode(node: Node): Xor[ParsingFailure, Json] = node match {
-    case anchor: AnchorNode => convertYamlNode(anchor.getRealNode)
-    case mapping: MappingNode =>
-      mapping.getValue.asScala.foldLeft(Right(JsonObject.empty) : Xor[ParsingFailure, JsonObject]) {
-        (objXor, tup) => for {
-          obj <- objXor
-          key <- convertKeyNode(tup.getKeyNode)
-          value <- convertYamlNode(tup.getValueNode)
-        } yield obj.add(key, value)
-      }.map(Json.fromJsonObject)
-    case sequence: SequenceNode =>
-      sequence.getValue.asScala.foldLeft(Right(List.empty[Json]) : Xor[ParsingFailure, List[Json]]) {
-        (arrXor, node) => for {
-          arr <- arrXor
-          value <- convertYamlNode(node)
-        } yield value :: arr
-      }.map(arr => Json.fromValues(arr.reverse))
-    case scalar: ScalarNode => convertScalarNode(scalar)
-  }
-
-  private def convertKeyNode(node: Node): Xor[ParsingFailure, String] = node match {
-    case scalar: ScalarNode => Right(scalar.getValue)
-    case _ => Left(ParsingFailure("Only string keys can be represented in JSON", null))
-  }
-
-  private def convertScalarNode(node: ScalarNode) = Xor.catchNonFatal(node.getTag match {
-    case ScalarTag("int") => Json.fromJsonNumber(JsonNumber.unsafeIntegral(node.getValue))
-    case ScalarTag("float") => Json.fromJsonNumber(JsonNumber.unsafeDecimal(node.getValue))
-    case ScalarTag("bool") => Json.fromBoolean(node.getValue.toBoolean)
-    case ScalarTag("null") => Json.Null
-    case CustomTag(other) =>
-      Json.fromJsonObject(JsonObject.singleton(other.stripPrefix("!"), Json.fromString(node.getValue)))
-    case other => Json.fromString(node.getValue)
-  }).leftMap {
-    err => ParsingFailure(err.getMessage, err)
-  }
-
-  private def parseYaml(reader: Reader) = for {
-    node <- Xor.catchNonFatal(new Yaml().compose(reader))
-    json <- convertYamlNode(node)
-  } yield json
-
-  private def parseYamlStream(reader: Reader) =
-    new Yaml().composeAll(reader).asScala.toStream.map(convertYamlNode)
-
-  def parse(reader: Reader) : Xor[ParsingFailure, Json] = parseYaml(reader).leftMap {
-    case err @ ParsingFailure(msg, underlying) => err
-    case err => ParsingFailure(err.getMessage, err)
-  }
-
-  def parse(string: String) : Xor[ParsingFailure, Json] = parse(new StringReader(string))
-
-  def parseDocuments(string: String) : Stream[Xor[ParsingFailure, Json]] = parseDocuments(new StringReader(string))
-  def parseDocuments(reader: Reader) : Stream[Xor[ParsingFailure, Json]] = parseYamlStream(reader)
-
-}
```

Comment:
I wasn't sure if I should leave these or not...

`Xor` was removed from both cats and circe in favor of using `Either`. This is probably because `Either` is becoming a right-favoring monad.

If we leave the code in place and add `Xor` back from an older version of cats, we risk unexpected issues from mixing older and newer versions of cats. If we switch from `Xor` to `Either`, then the code is not binary compatible anyway, so why not just have the users of the API switch over to the new location for the code.

My thought in the end was to remove it, and make this a major upgrade. We can backport some quick fixes and deprecation warnings. What do you think @jeremyrsmith?